parlai/tasks/wizard_of_wikipedia/agents.py

```diff
@@ -344,33 +344,41 @@ class BasicdialogTeacher(WizardOfWikipediaTeacher):
     def __init__(self, opt, shared=None):
         super().__init__(opt, shared)
         self.num_exs = sum(len(d['dialog']) // 2 for d in self.data)
-        self.wizard_dialog = opt.get('wizard_dialog', False)
+        self.speaker_label = opt['speaker_label']
 
     @staticmethod
     def add_cmdline_args(argparser):
         agent = argparser.add_argument_group('Basic Dialog Arguments')
         agent.add_argument(
-            '--wizard-dialog',
-            type='bool',
-            default=False,
-            help='If true, ensures that wizard response ' 'is always the label',
+            '--speaker-label',
+            type=str,
+            default='both',
+            choices=['both', 'wizard', 'apprentice'],
+            help='Which speaker labels to train on',
         )
 
     def num_examples(self):
         return self.num_exs
 
     def len_episode(self, ep):
         d = self.data[ep]
-        if self.wizard_dialog and ('Wizard' in d['dialog'][0]['speaker']):
-            return (len(d['dialog']) - 1) // 2
+        if self.speaker_label != 'both':
+            first_speaker = d['dialog'][0]['speaker']
+            if self.speaker_label == 'wizard' and 'Wizard' in first_speaker:
```

Comment:
nit: since these two conditions do the same thing I feel like we could probably combine them more elegantly but it's also not gonna look too great if we just do "A and B or C and D" so up to you

Reply:
yeah I thought the line was gonna be too long and it would look bad lol

Reply:
although...maybe not, i'll try